src/hooks/useWearables.ts

```diff
@@ -0,0 +1,114 @@
+import { useQuery } from 'react-query';
+import {
+  SyncTypeSettings,
+  ToggleWearableResult,
+  WearableIntegrationFailureCode,
+  WearableIntegrationStatus,
+  WearablesSyncState,
+} from '../components/Wearables/WearableTypes';
+import { useActiveAccount } from './useActiveAccount';
+import { useHttpClient } from './useHttpClient';
+
+/**
+ *
+ * @returns UTC offset in ±HHMM format, e.g., -04:00 for New York City during daylight saving time
+ * or -05:00 during standard time. Equivalent to moment's utcOffset() function.
+ */
+const getUTCOffset = () => {
+  const offsetInMinutes = new Date().getTimezoneOffset();
+  const sign = offsetInMinutes > 0 ? '-' : '+';
+  const hours = Math.abs(Math.floor(offsetInMinutes / 60))
+    .toString()
+    .padStart(2, '0');
+  const minutes = Math.abs(offsetInMinutes % 60)
+    .toString()
+    .padStart(2, '0');
+  const offset = `${sign}${hours}${minutes}`;
+  return offset;
+};
+
+export const useWearableIntegrations = (appId: string) => {
+  const { httpClient } = useHttpClient();
+  const { accountHeaders } = useActiveAccount();
+  return useQuery('/v1/wearables', () =>
+    httpClient
+      .get<WearablesSyncState>('/v1/wearables', {
+        params: { appId },
+        headers: accountHeaders,
+      })
+      .then((res) => res.data),
+  );
+};
+
+export const useWearableIntegration = (ehrId: string) => {
+  const { httpClient } = useHttpClient();
+  const { accountHeaders } = useActiveAccount();
+  return useQuery('/v1/wearables/{ehrId}', () => {
+    httpClient.get(`/v1/wearables/${ehrId}`, { headers: accountHeaders });
+  });
+};
+
+export const useSetSyncTypes = () => {
+  const { httpClient } = useHttpClient();
+  const { accountHeaders } = useActiveAccount();
+  const setSyncTypes = (settings: SyncTypeSettings) =>
+    httpClient.put('/v1/wearables/sync-types', settings, {
+      headers: accountHeaders,
+    });
+  return setSyncTypes;
```

Comment:
Did you consider using `useMutation` from `react-query` and decide against it?  While `useQuery` can be a little tougher to get right, I think it might be interesting to refactor this and the 2 methods below it into a single hook which exposes 3 async callbacks.  That'd allow for `httpClient` and `accountHeaders` sharing too.

Reply:
That would be interesting, I'd like to see how that would look too. I will try it out in my follow up PR.